ci-operator/step-registry/redhat-developer/rhdh/redhat-developer-rhdh-commands.sh

```diff
@@ -82,8 +83,29 @@ for change in $PR_CHANGESET; do
         ONLY_IN_DIRS=false
         break
     fi
+    # Check if there were any changes in .ibm/images/
+    if echo "$change" | grep -qE "^.ibm/images/"; then
+        IMAGE_BUILD_NEEDED=true
+        break
+    fi
 done
 
+# If no changes are detected in .ibm/images/, skip the image build process
+if [ "$IMAGE_BUILD_NEEDED" = false ]; then
+    echo "No changes detected in .ibm/images/. Skipping showcase-e2e-runner image build..."
+else
+    echo "Changes detected in .ibm/images/. Starting image build..."
+
+    # Build the Docker image
+    podman build -t quay.io/rhdh-community/rhdh-e2e-runner:latest .ibm/images/
+
+    # Authenticate with Quay.io
+    podman login -u "${QUAY_USERNAME}" -p "${QUAY_PASSWORD}" quay.io
+
+    # Push the newly built image to Quay.io
+    podman push quay.io/rhdh-community/rhdh-e2e-runner:latest
```

Comment:
This will lead to a PR job replacing the image before we merge the new version of the Dockerfile.
Let's say I open a PR with changes in the Docker file, run the PR check, and the `podman push` would replace the image for all the other jobs running after that PR check. 
We should use a different logic here and/or use a GH action.

Reply:
@zdrapela To resolve this, a full refactor of the pipeline was required, introducing a new post-merge job and a dynamic image selection strategy.